source/addonHandler/__init__.py

```diff
@@ -459,6 +469,13 @@ def _addonGuiModel(self) -> "AddonManifestModel":
 		from _addonStore.models.addon import _createGUIModelFromManifest
 		return _createGUIModelFromManifest(self)
 
+	@property
+	def _hasOverriddenCompat(self) -> bool:

```

Comment:
This property should probably be added to `SupportsVersionCheck` like similar properties

I think this property might easily get confused with `overrideIncompatibility`, which now is only used when compatibility should already be overriden, not pending override.

```suggestion
	def _hasOverriddenCompat(self) -> bool:
		"""If True, this add-on has been manually overriden. The affects of override may be pending restart"""
```

Reply:
It would be worth adding a docstring to `overrideIncompatibility` as well

> If `True` NVDA should enable this add-on where it would normally be blocked due to incompatibility

Reply:
I've added docstrings you have suggested.
As to moving the property into `SupportsVersionCheck`, that was also my first thought when writing this PR, but now I'm not too sure about it. I placed it in the `AddonBase since:
1. It is explicitly used only in `AddonBase` subclasses
2. Adding it into `AddonBase` makes boundaries between `addonHandler` and add-on store code more clear, thoug this is probably somewhat subjective
3. I cannot come with an use case in which it would have to be called against other subclasses of `SupportsVersionCheck`, and since this is private code we can always move it if the need arises

Reply:
I'm not sure I understand the boundaries argument, as `_hasOverriddenCompat` is used in `addonStore`.
Logically grouping it with other version related checks is more useful in my opinion.
I think long term we want to whittle down and migrate addonHandler and addonGui code to addonStore and addonStoreGui.

Reply:
There's a huge number of properties for various add-on datastructures now. 
There were 2 dev approaches to simplify them:

- Properties based on context (bundle, installed add-on, available from add-on store)
- Properties based on utility (version checking, status checking)